apis/v1beta1/gateway_types.go

```diff
@@ -24,7 +24,6 @@ import (
 // +kubebuilder:object:root=true
 // +kubebuilder:resource:categories=gateway-api,shortName=gtw
 // +kubebuilder:subresource:status
-// +kubebuilder:storageversion
```

Comment:
Just to confirm, we're leaving the storage version at v1alpha2 for now, and will migrate it later? This makes sense to me, but is probably worth calling out.